storage/tianmu/handler/tianmu_handler.cpp

```diff
@@ -168,7 +170,17 @@ namespace {
 std::vector<bool> GetAttrsUseIndicator(TABLE *table) {
   int col_id = 0;
   std::vector<bool> attr_uses;
+  enum_sql_command sqlCommand = SQLCOM_END;
+  if (table->in_use && table->in_use->lex)
+    sqlCommand = table->in_use->lex->sql_command;
+  bool tianmuDeleteOrUpdate = sqlCommand == SQLCOM_DELETE || sqlCommand == SQLCOM_DELETE_MULTI ||
+                              sqlCommand == SQLCOM_UPDATE || sqlCommand == SQLCOM_UPDATE_MULTI;
   for (Field **field = table->field; *field; ++field, ++col_id) {
+    // Binlog in row format needs to fill all columns
```

Comment:
Why we should add all columns into? the root cause is what. pls gives your explanation. this needs more words to explain it.